### PR TITLE
chore: (TNLT-6506/TNLT-6563) New Ad logic

### PR DESCRIPTION
**Pull request by Geoff-Ford**
https://nidigitalsolutions.jira.com/browse/TNLT-6506
https://nidigitalsolutions.jira.com/browse/TNLT-6563

* Enables adverts on In Depth and Magazine Standard templates.
* Implements the following rules/logic to determine when and what type of advert to display:
  * If an article has less than 6 paragraphs of text then do not show any advert
  * If an article has 6 or 7 paragraphs of text then show a single mpu ad aligned the start of the 5th paragraph
  * If an article has 8 or more paragraphs of text, but also includes inline images/interactives in the first 10 items then show a single mpu
  * If an article has 8 or more paragraphs of text but does not include any inline images/interactives in the first 10 items then show a double mpu
  * If an article has an image/interactive directly above and below the 5th paragraph (i.e. top ten countries to visit) then show a leaderboard directly below the 5th paragraph
* Updated/added tests to cover the additional cases

### Enables adverts on In Depth template
6 - Ads enabled on indepth template

### Enables adverts on Magazine Standard template
5 - Ads enabled on magazinestandard template

### If an article has less than 6 paragraphs of text then do not show any advert
1 - No ads if less than 6

### If an article has 6 or 7 paragraphs of text then show a single mpu ad aligned the start of the 5th paragraph
2 - Single MPU if less than 8

### If an article has 8 or more paragraphs of text, but also includes inline images/interactives in the first 10 items then show a single mpu
3 - Single MPU if 8 or more and has non paragraph

### If an article has 8 or more paragraphs of text but does not include any inline images/interactives in the first 10 items then show a double mpu
4 - Double MPU if 8 or more and no non paragraph

##### Before
7 - Before - Portrait - Leaderboard when 5th para preceded and followed by non para

##### After
7 - After - Portrait - Leaderboard when 5th para preceded and followed by non para

##### Before
8 - Before - Landscape - Billboard when 5th para preceded and followed by non para

##### After
8 - After - Landscape - when 5th para preceded and followed by non para

### CODE REVIEW

File: packages/article-skeleton/__tests__/body-utils/setup-ad.js [hunk 1]
```diff
@@ -73,22 +73,130 @@ export default () => {
...
+      const newSkeletonProps = {
+        ...skeletonProps,
+        data: { ...skeletonProps.data, content: longContent },
+      };
+
+      expect(setupAd(newSkeletonProps)).toEqual([
+        createParagraph("a"),
+        createParagraph("b"),
+        createParagraph("c"),
+        createParagraph("d"),
```
Comment (c-j-j) [inline packages/article-skeleton/__tests__/body-utils/setup-ad.js:164]: I was expecting this output to have the ad after 6 paragraphs, but it's after the 4th paragraph here? Also where's paragraphs `e`,`f` and `g`?

File: packages/article-skeleton/src/article-body/article-body-row.js [hunk 2]
```diff
@@ -175,6 +175,7 @@ export default ({
       );
     },
     ad(key, attributes) {
+      console.log("DASDASDASDASDASDASASADASAAAAAAAAAA", attributes);
```
Comment (c-j-j) [inline packages/article-skeleton/src/article-body/article-body-row.js:178]: this needs deleting

Comment (kamiranoff) [inline packages/article-skeleton/src/article-body/article-body-row.js:178]: I guess this console.log was not that easy to see after all.

File: packages/article-skeleton/__tests__/body-utils/setup-ad.js [hunk 3]
```diff
@@ -123,27 +232,26 @@ export default () => {
           children: [],
         },
         createParagraph("l"),
-        { name: "image", children: [] },
         createParagraph("m"),
       ]);
     });
 
-    it("should return content with the inline single mpu ad present when non-paragraph content within threshold", () => {
+    it("should return content with an inline double mpu ad present and attributes overriden when more than 8 and no non-paragraph content within threshold", () => {
```
Comment (aronshamash) [inline packages/article-skeleton/__tests__/body-utils/setup-ad.js:239]: thats a mouthful!

**Review by aronshamash — APPROVED** [review]
looks good and logic is easy to follow

File: packages/article-skeleton/src/article-body/article-body-row.js (hunk 2 above)
Comment (Geoff-Ford) [reply packages/article-skeleton/src/article-body/article-body-row.js:178]: haha

**Review by kamiranoff — APPROVED** [review]

File: packages/article-skeleton/__tests__/body-utils/setup-ad.js (hunk 3 above)
Comment (Geoff-Ford) [reply packages/article-skeleton/__tests__/body-utils/setup-ad.js:239]: Well don't eat it then

**Review by c-j-j — APPROVED** [review]
One comment about the tests but all looks good... definitely provides better UX this way

File: packages/article-skeleton/__tests__/body-utils/setup-ad.js (hunk 1 above)
Comment (Geoff-Ford) [reply packages/article-skeleton/__tests__/body-utils/setup-ad.js:164]: That is the correct position of the ad i.e. the 5th paragraph (and beyond if necessary) gets inlined with the ad.

e, f, and g have been inlined (see below)

Comment (c-j-j) [reply packages/article-skeleton/__tests__/body-utils/setup-ad.js:164]: Ah I misread the test name... this is testing when there's 6 or 7 paragraphs total. I thought the ad was going `between 6 and 7 paragraphs`. Ignore

File: packages/article-skeleton/src/article-body/article-body-row.js (hunk 2 above)
Comment (Geoff-Ford) [reply packages/article-skeleton/src/article-body/article-body-row.js]: Done
Screenshot 2021-02-04 at 16 47 55

Comment (Geoff-Ford) [reply packages/article-skeleton/src/article-body/article-body-row.js]: Bah! and it isn't even a comment!

Comment (kamiranoff) [reply packages/article-skeleton/src/article-body/article-body-row.js]: ahah

**Pull request merged**